src/lib.rs

```diff
@@ -1052,17 +1052,18 @@ impl<'a, L: 'static + Ledger> KeystoreState<'a, L> {
         block_id: u64,
         txn_id: u64,
         kind: TransactionKind<L>,
-        txn_hash: TransactionHash<L>,
+        hash: TransactionHash<L>,
         records: &[RecordOpening],
     ) {
-        let history = receive_history_entry(kind, txn_hash, records);
+        let uid = TransactionUID::<L>(hash.clone());
+        let history = receive_history_entry(kind, uid, records);
 
         if let Err(err) = model
-            .store(|mut t| async move {
-                t.store_transaction(history).await?;
-                Ok(t)
-            })
-            .await
+            .transactions
+            .create(history)
+            .unwrap()
```

Comment:
Definitely should not be unwrapping here. This will panic if there's a temporary error in persistence. Just fall through to the println for now.

(This is another thing that's improved in #231, which allows this function to return an error)